ls1intum/Artemis
# Allow instructors to enable / disable the offline IDE usage in programming exercises

Fixes #1661 




### Checklist

### Motivation and Context
Instructors might want to create programming exercises where students can only use the online code editor and disallow the use of the offline IDE (and git). This was not possible until now.

### Description
I added a new option to allow the offline IDE usage (by default this is true).
In case this option is set to false by the instructor, the students will not get access to the git repository when they start the exercise and they won't see the clone repository button.

### Steps for Testing
1. Create a programming exercise with "Allow offline IDE" disabled
2. Start the programming exercise
3. Double check that the clone repository button is never visible
4. (You should also not be able to import this exercise into Orion)
5. Double check that you do not have access rights to the git repository
6. Change the configuration to "Allow offline IDE" enabled
7. Delete your participation for this programming exercise (including its repository)
8. Start the exercise again
9. Now the clone button should be visible, you should have access rights in Bitbucket and you should be able to clone the repository

Note: there are some limitations when instructors change the value after some participations have been create (i.e. students started to work on the programming exercise). Then the access for existing repositories is not changed. This is out of scope for this PR and should be improved in a future PR.
